website/pages/guides/integrations/with-gatsby.mdx

```diff
@@ -69,17 +69,17 @@ export default theme
 Chakra exposes some context providers to give your website context of the theme,
 and color mode.
 
-Create the two files [`gatsby-browser.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#wrapRootElement) and [`gatsby-ssr.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-ssr/#wrapRootElement) at the root of your Gatsby project.
+Create the two files [`gatsby-browser.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#wrapRootElement) and [`gatsby-ssr.js`](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-ssr/#wrapRootElement) at the root of your Gatsby project. Also create a `provider.js` that contains a React component that will be used in both files.
```

Comment:
It's been awhile since I've used `gatsby`, so trying to reason through this.

If the plugin already contains `gatsby-browser.js` and `gatsby-ssr.js` and the user can provide options via `gatsby-config.js`, would there be a need to do any of this? 

Could we maybe allow users to import the plugin's `WrapRootElement` if they wanted to render other things inside of it as part of the wrapper, so they wouldn't have to duplicate the effort of setting up `ChakraProvider` that we already do inside the plugin?

Reply:
> If the plugin already contains gatsby-browser.js and gatsby-ssr.js and the user can provide options via gatsby-config.js, would there be a need to do any of this?

If the plugin exposes everything you can configure on `ChakraProvider` then there's not really the need to set up providers manually.

Gatsby merges the APIs together when multiple plugins use the same API so when a user uses `wrapRootElement` API in their e.g. `gatsby-ssr.js` file they don't need to import the plugin part or even care about it.